exchanges/ftx/ftx_wrapper.go

```diff
@@ -0,0 +1,871 @@
+		orderBook.Asks = append(orderBook.Asks, orderbook.Item{
+			Amount: tempResp.Asks[y].Size,
+			Price:  tempResp.Asks[y].Price})
+	}
+	orderBook.Pair = p
+	orderBook.ExchangeName = f.Name
+	orderBook.AssetType = assetType
+	err = orderBook.Process()
+	if err != nil {
+		return orderBook, err
+	}
+	return orderbook.Get(f.Name, p, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) UpdateAccountInfo() (account.Holdings, error) {
+	var resp account.Holdings
+	data, err := f.GetBalances()
+	if err != nil {
+		return resp, err
+	}
+	var acc account.SubAccount
+	for i := range data.Result {
+		c := currency.NewCode(data.Result[i].Coin)
+		hold := data.Result[i].Total - data.Result[i].Free
+		total := data.Result[i].Total
+		acc.Currencies = append(acc.Currencies,
+			account.Balance{CurrencyName: c,
+				TotalValue: total,
+				Hold:       hold})
+	}
+	resp.Accounts = append(resp.Accounts, acc)
+	resp.Exchange = f.Name
+
+	err = account.Process(&resp)
+	if err != nil {
+		return account.Holdings{}, err
+	}
+
+	return resp, nil
+}
+
+// FetchAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) FetchAccountInfo() (account.Holdings, error) {
+	acc, err := account.GetHoldings(f.Name)
+	if err != nil {
+		return f.UpdateAccountInfo()
+	}
+
+	return acc, nil
+}
+
+// GetFundingHistory returns funding history, deposits and
+// withdrawals
+func (f *FTX) GetFundingHistory() ([]exchange.FundHistory, error) {
+	var resp []exchange.FundHistory
+	depositData, err := f.FetchDepositHistory()
+	if err != nil {
+		return resp, err
+	}
+	for x := range depositData.Result {
+		var tempData exchange.FundHistory
+		tempData.Fee = depositData.Result[x].Fee
+		tempData.Timestamp = depositData.Result[x].Time
+		tempData.ExchangeName = f.Name
+		tempData.CryptoTxID = depositData.Result[x].TxID
+		tempData.Status = depositData.Result[x].Status
+		tempData.Amount = depositData.Result[x].Size
+		tempData.Currency = depositData.Result[x].Coin
+		tempData.TransferID = strconv.FormatInt(depositData.Result[x].ID, 10)
+		resp = append(resp, tempData)
+	}
+	withdrawalData, err := f.FetchWithdrawalHistory()
+	if err != nil {
+		return resp, err
+	}
+	for y := range withdrawalData.Result {
+		var tempData exchange.FundHistory
+		tempData.Fee = depositData.Result[y].Fee
+		tempData.Timestamp = depositData.Result[y].Time
+		tempData.ExchangeName = f.Name
+		tempData.CryptoTxID = depositData.Result[y].TxID
+		tempData.Status = depositData.Result[y].Status
+		tempData.Amount = depositData.Result[y].Size
+		tempData.Currency = depositData.Result[y].Coin
+		tempData.TransferID = strconv.FormatInt(depositData.Result[y].ID, 10)
+		resp = append(resp, tempData)
+	}
+	return resp, nil
+}
+
+// GetExchangeHistory returns historic trade data since exchange opening.
+func (f *FTX) GetExchangeHistory(p currency.Pair, a asset.Item, timestampStart, timestampEnd time.Time) ([]exchange.TradeHistory, error) {
```

Comment:
Need to change the signature of all exchanges to implement the IBotExchange interface.